### PR TITLE
[Artist] implements A/B/C/D variants of artist page

### PR DESCRIPTION
**Pull request by l2succes**
Zeplin: https://zpl.io/aXWXPxg

### CODE REVIEW

File: src/desktop/apps/article/client/classic.js [hunk 1]
```diff
@@ -30,7 +33,7 @@ export const init = () => {
 
   new ArticleView({
     el: $('body'),
-    article
+    article,
```
Comment (cavvia) [inline src/desktop/apps/article/client/classic.js]: why are these trailing commas needed?

**Comment by cavvia** [comment]
I'm finding it difficult to review your changes here due to the high volume of style changes - the vast majority of this diff is style related and the diff is pretty large (56 edited files). It would be easier if you PR'd the style changes separately.

File: src/desktop/components/split_test/running_tests.coffee [hunk 2]
```diff
@@ -15,10 +15,24 @@
 # module.exports = {}
 
 module.exports = {
- onboarding_test:
-   key: 'onboarding_test'
-   outcomes:
-     control: 50
-     experiment: 50
-   edge: 'experiment'
+  collect_keyword_search:
```
Comment (cavvia) [inline src/desktop/components/split_test/running_tests.coffee]: this test has been completed and should be removed from here i believe - a rebase with master should have removed it.

File: src/desktop/components/split_test/running_tests.coffee [hunk 3]
```diff
@@ -15,10 +15,24 @@
...
+      control: 50
+      experiment: 50
+    edge: 'experiment'
+  artist_page:
+    key: 'artist_page'
+    outcomes:
+      control: 25
+      no_info: 25
+      no_header: 25
+      no_info_header: 25
```
Comment (cavvia) [inline src/desktop/components/split_test/running_tests.coffee]: looks good :+1:

**Comment by alloy** [comment]
@cavvia It looks like the style changes are only in the ‘merge master’ commit, the other two commits are pretty clean.

As to why trailing commas are added, that’s a convention described [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas), our prettier setups will add these by default.

**Comment by cavvia** [comment]
Yes, thanks @alloy - I had a look at the separate commits which helped. I don't quite understand why a merge with master would result in all these style additions, but maybe that's just a misleading commit message.

@l2succes I've added a couple of front-end devs to the review list as I'm largely unfamiliar with Force.

**Comment by l2succes** [comment]
@cavvia agreed, sorry about that, I had an issue while fixing merge conflicts in this branch. I removed all the styling updates from this PR

**Comment by cavvia** [comment]
Thanks @l2succes! My only outstanding issue is regarding the keyword search A/B test configuration, which I don't believe should be added here.

File: src/desktop/apps/artist/routes.coffee [hunk 4]
```diff
@@ -36,6 +38,9 @@ sd = require('sharify').data
 
       return res.redirect(artist.href) unless(_.find nav.sections(), slug: tab) or artist.counts.artworks is 0
 
+      artistTest = new SplitTest req, res, runningTests.artist_page
```
Comment (mzikherman) [inline src/desktop/apps/artist/routes.coffee]: I don't think you need to do this. By setting up the split test, you automatically create a sharify var that will contain the value.

so in this case, `sd.ARTIST_PAGE` can be checked against 'control', etc.

So you don't have to finagle it here. I might recommend changing the name, `sd.ARTIST_PAGE` is not super descriptive.

File: src/desktop/apps/artist/client/views/overview.coffee [hunk 5]
```diff
@@ -125,10 +125,15 @@ module.exports = class OverviewView extends Backbone.View
 
   render: ->
     # Template expects plain JSON, not a Backbone model.
+    outcome = splitTest('artist_page')?.outcome()
```
Comment (mzikherman) [inline src/desktop/apps/artist/client/views/overview.coffee]: Same as below comment, you already get this value via `sd.ARTIST_PAGE`.

File: src/desktop/components/split_test/running_tests.coffee [hunk 6]
```diff
@@ -15,10 +15,18 @@
...
+    edge: 'experiment'
```
Comment (mzikherman) [inline src/desktop/components/split_test/running_tests.coffee]: The `edge` is what admins get sent to, so it should be one of the 4 test options.

File: src/desktop/apps/artist/routes.coffee (hunk 4 above)
Comment (l2succes) [reply src/desktop/apps/artist/routes.coffee]: Got it, thanks for the heads up. I'll update the name as well

File: src/desktop/apps/artist/client/views/overview.coffee [hunk 7]
```diff
@@ -125,10 +125,15 @@ module.exports = class OverviewView extends Backbone.View
...
   render: ->
     # Template expects plain JSON, not a Backbone model.
+    testGroup = sd.ARTIST_PAGE_VARIANTS
+
     @$el.html template
       artist: @model.toJSON()
       viewHelpers: viewHelpers
       statuses: @statuses
+      showSections:
+        heder: testGroup is 'control' or testGroup is 'no_info'
```
Comment (mzikherman) [inline src/desktop/apps/artist/client/views/overview.coffee]: typo **header**

Comment (l2succes) [reply src/desktop/apps/artist/client/views/overview.coffee]: ha! good catch.. gracias

**Comment by cavvia** [comment]
It looks like all the comments have been addressed - is this good to go now @mzikherman?

**Comment by mzikherman** [comment]
Yes looks good!

**Comment by cavvia** [comment]
@l2succes If this switches on the A/B test - which I _think_ it does - we need to park it till we have the OK from analytics. In which case we shouldn't be merging it into master just yet.

**Comment by mzikherman** [comment]
Gah, my bad! I got merge happy.

**Comment by cavvia** [comment]
No worries! Going to revert until the analytics work is in place.

**Comment by l2succes** [comment]
Yes, let's revert until we get the OK from analytics.

**Pull request merged**